### PR TITLE
optimize: build the frontend at the same time as the maven build

### PR DESCRIPTION
**Pull request by liuqiufeng**
- [ ] I have registered the PR [changes](../changes).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -53,3 +53,8 @@ Thumbs.db
 
 #h2
 *.db
+/console/src/main/resources/static/css/
```
Comment (slievrly) [inline .gitignore:56]: why ignore these files?

File: console/pom.xml [hunk 2]
```diff
@@ -126,11 +126,80 @@
...
+                    <workingDirectory>src/main/resources/static/console-fe</workingDirectory>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>install node and npm</id>
+                        <goals>
+                            <goal>install-node-and-npm</goal>
+                        </goals>
+                        <configuration>
+                            <nodeVersion>v8.16.0</nodeVersion>
```
Comment (slievrly) [inline console/pom.xml]: can we use v19.5.0 ?

File: console/pom.xml [hunk 3]
```diff
@@ -126,11 +126,80 @@
...
+                            <arguments>install</arguments>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>update caniuse db</id>
+                        <goals>
+                            <goal>npx</goal>
+                        </goals>
+                        <configuration>
+                            <arguments>browserslist@latest --update-db</arguments>
```
Comment (slievrly) [inline console/pom.xml:164]: if necessary?

**Review by slievrly — COMMENTED** [review]
Need to consider the independent deployment of console.

File: .gitignore (hunk 1 above)
Comment (liuqiufeng) [reply .gitignore:56]: Because these files are generated together in the maven package.

File: console/pom.xml (hunk 2 above)
Comment (liuqiufeng) [reply console/pom.xml]: incompatible

File: console/pom.xml (hunk 3 above)
Comment (liuqiufeng) [reply console/pom.xml:164]: Not necessary, but recommended.

**Comment by liuqiufeng** [comment]
> Need to consider the independent deployment of console.

None of these changes affect the separate packaging of the console front-end

**Review by slievrly — DISMISSED** [review]
LGTM

File: console/pom.xml [hunk 4]
```diff
@@ -126,11 +126,80 @@
...
     </dependencies>
     <build>
+        <plugins>
+            <plugin>
+                <groupId>com.github.eirslett</groupId>
+                <artifactId>frontend-maven-plugin</artifactId>
+                <configuration>
+                    <workingDirectory>src/main/resources/static/console-fe</workingDirectory>
+                </configuration>
+                <executions>
```
Comment (slievrly) [inline console/pom.xml:136]: any phase here ?

Comment (liuqiufeng) [reply console/pom.xml:136]: > any phase here ?

Added phase here.

File: console/pom.xml [hunk 5]
```diff
@@ -166,6 +169,7 @@
                         <goals>
                             <goal>npm</goal>
                         </goals>
+                        <phase>generate-resources</phase>
```
Comment (slievrly) [inline console/pom.xml:172]: it is unreasonable to set it in the generate-resources phase.

Comment (slievrly) [reply console/pom.xml:172]: Considering online IDE debugging, maintain the status quo.

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**